src/cell_list.h

```diff
@@ -0,0 +1,191 @@
+                ++current_distance;
+                if( current_distance > distance_to_nearest_cell_wall ) {
+                    load();
+                }
+                current = sorted_elements[current_distance].begin();
+            }
+            if( current_distance > range->distance ) {
+                range = nullptr;
+            }
+            return *this;
+        }
+
+        bool operator==( const cell_list_iterator &rhs ) const {
+            if( range == nullptr && rhs.range == nullptr) {
+                return true;
+            }
+            if( range != rhs.range ) {
+                return false;
+            }
+            return current == rhs.current;
+        }
+        bool operator!=( const cell_list_iterator &rhs ) const {
+            return !(*this == rhs);
+        }
+
+    private:
+        void load() {
+            // Need another rank loaded.
+            int min_cell_x = range->origin_cell.x - cell_ranks_loaded;
+            int max_cell_x = range->origin_cell.x + cell_ranks_loaded + 1;
+            int min_cell_y = range->origin_cell.y - cell_ranks_loaded;
+            int max_cell_y = range->origin_cell.y - cell_ranks_loaded + 1;
+            int minx = range->distance;
+            if( min_cell_x >= 0 ) {
+                minx = range->origin.x - CELL_SIZE * min_cell_x;
+            }
+            int maxx = range->distance;
+            if( max_cell_x <= MAXX / CELL_SIZE ) {
+                maxx = CELL_SIZE * ( max_cell_x + 1 ) - 1 - range->origin.x;
+            }
+            int miny = range->distance;
+            if( min_cell_y >= 0 ) {
+                miny = range->origin.y - CELL_SIZE * min_cell_y;
+            }
+            int maxy = range->distance;
+            if( max_cell_y <= MAXY / CELL_SIZE ) {
+                maxy = CELL_SIZE * ( max_cell_y + 1 ) - 1 - range->origin.y;
+            }
+            int min_x_dist = std::min( minx, maxx );
+            int min_y_dist = std::min( miny, maxy );
+            int min_dist = std::min( min_x_dist, min_y_dist );
+            min_dist = std::min( min_dist, range->distance );
+            distance_to_nearest_cell_wall = min_dist;
+            if( cell_ranks_loaded == 0 ) {
+                // Special case for first cell.
+                for( const std::pair<point, T *> &candidate : range->list->cells[min_cell_y][min_cell_x] ) {
+                    const int distance = rl_dist( candidate.first, range->origin );
+                    if( distance > range->distance ) {
+                        continue;
+                    }
+                    sorted_elements[distance].push_back( candidate );
+                }
+            } else {
+                // Draw a box...
+                for( int x = min_cell_x; x < max_cell_x; ++x ) {
+                    for( int y = min_cell_y; y < max_cell_y; ++y ) {
+                        for( const std::pair<point, T *> &candidate : range->list->cells[min_cell_y][min_cell_x] ) {
```

Comment:
```suggestion
                        for( const std::pair<point, T *> &candidate : range->list->cells[y][x] ) {
```
I'm guessing this is what was intended?

Reply:
This whole chunk is a placeholder, so yea that's closer, but I just put it all up for review because I was stuck on a dead end with the vector initialization thing, which turned out to be a red herring.